### PR TITLE
Make namespace of PrometheusRule configurable 

### LINKED ISSUE #43: Make PrometheusRule Namespace configurable
**Is your feature request related to a problem? Please describe.**
By default, and without customizations, Prometheus only looks for PrometheusRules and ServiceMonitors in its own Namespace. Currently, the ServiceMonitor Namespace is configurable but the [PrometheusRule](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/jenkins-master-alerting-rules.yaml#L7) is not.

**Describe the solution you'd like**
Add support to allow PrometheusRule to be configured via `.Values.master.prometheus.prometheusRuleNamespace`

**Example:**
```helm
{{- if .Values.master.prometheus.prometheusRuleNamespace }}
  namespace: {{ .Values.master.prometheus.prometheusRuleNamespace }}
{{- else }}
  namespace: {{ template "jenkins.namespace" . }}
{{- end }}
```

Also, add these to the `values.yaml` file and Document them.

**Describe alternatives you've considered**
Modifying Prometheus to scan other Namespaces

**Additional context**
Copied from [#23306](https://github.com/helm/charts/issues/23306)

### PR DESCRIPTION
**Pull request by prabhu43**
Signed-off-by: Prabhu Jayakumar <j.prabhu91@gmail.com>

# What this PR does / why we need it
It makes namespace of PrometheusRule configurable.

# Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #43 

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/values.yaml [hunk 1]
```diff
@@ -476,6 +476,8 @@ master:
     # See here: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
     # The `groups` root object is added by default, simply add the rule entries
     alertingrules: []
+    # Set a custom namespace where to deploy PrometheusRule resource
+    # prometheusRuleNamespace: monitoring
```
Comment (torstenwalter) [inline charts/jenkins/values.yaml:480]: Can we uncomment it and use empty string as default? That enables auto completion in some IDEs.

**Review by torstenwalter — CHANGES_REQUESTED** [review]

File: charts/jenkins/values.yaml (hunk 1 above)
Comment (prabhu43) [reply charts/jenkins/values.yaml]: @torstenwalter i changed it as you suggested. please check now

Comment (torstenwalter) [reply charts/jenkins/values.yaml]: Looks good to me. Can you merge main into this branch and fix the merge conflicts?

Comment (prabhu43) [reply charts/jenkins/values.yaml]: done

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**